backend/src/main/java/com/htecgroup/skynest/util/JwtUtils.java

```diff
@@ -58,13 +66,49 @@ public static UsernamePasswordAuthenticationToken getFrom(String token) {
     }
   }
 
+  public static String generateEmailToken(String email, String purpose) {
+    return JWT.create()
+        .withSubject(email)
+        .withExpiresAt(new Date(System.currentTimeMillis() + EMAIL_TOKEN_EXPIRATION_MS))
+        .withClaim(EMAIL_TOKEN_CLAIM, purpose)
+        .sign(ALGORITHM);
+  }
+
+  public static String validateEmailToken(String token, String purpose) {
+    try {
+      Verification verification = JWT.require(ALGORITHM);
+      JWTVerifier verifier = verification.withClaim(EMAIL_TOKEN_CLAIM, purpose).build();
+      DecodedJWT decodedJWT = verifier.verify(token);
+      return decodedJWT.getSubject();
+    } catch (JWTVerificationException e) {
+      log.error("Invalid JWT token: {}", e.getMessage());
+      throw new UserException(UserExceptionType.INVALID_TOKEN);
+    } catch (IllegalArgumentException e) {
+      log.error("JWT claims string is empty: {}", e.getMessage());
+      throw new UserException("Access token is missing", HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  public static String validatePasswordResetToken(String token) {
+    return validateEmailToken(token, JwtUtils.PASSWORD_RESET_PURPOSE);
+  }
+
+  public static String validateEmailVerificationToken(String token) {
+    return validateEmailToken(token, JwtUtils.EMAIL_VERIFICATION_PURPOSE);
+  }
+
   @Value("${jwt.access-expiration-ms}")
-  public void setNameStatic(long expirationMs) {
+  private void setAccessTokenExpirationMs(long expirationMs) {
```

Comment:
you can always set that value in field instead of method, meaning 
```
@Value("${jwt.access-expiration-ms}")
private final Long ACCESS_TOKEN_EXPIRATION_MS;
```

I think that was the correct syntax

Reply:
I tried that first but it doesn't work with static fields. So I won't be able to use the variable it in static methods